### DISCUSSION TITLE
[portable] Is it safe to store databases in D:/keepassxc/data/base.kdbx?

### DISCUSSION THREAD

**Discussion by TolyanDimov** [keepassxreboot/keepassxc#8926] [Q&A] [answered]
I'm not sure if it's safe to store base.kdbx inside the directory?

My actions:
Unpacked the **Portable** version to a USB device.
I created a new folder inside the directory with the name `data`, and put my base there `base.kdbx`.
Full Path: `D:\KeePassXC\data\base.kdbx`

Will the directory be overwritten, for example, when updating and will it delete my database?

Thank you in advance!

_Printed by Yandex.Translate._

**Comment by droidmonkey** [comment, marked as answer]
Unfortunately that's a really great way to accidentally delete your database. We also do not recommend you store your database on A USB drive. There's a high risk of corruption.

**Reply by TolyanDimov** [reply]
Thanks for your reply. You made me think.

I think the ideal password storage scheme is to learn them by heart. But there are more and more passwords, and this forces me to use KeePassXC.
I know that flash drives are insecure, and keys need to be stored separately from the database, but this is not always convenient.

**Reply by michaelk83** [reply]
Not only are the more passwords, but human generated ones tends to be weak. Your best bet is to use the built-in password generator to make completely random passwords (those are not designed to be remembered). If there are some you want to remember, switch the generator to the passphrase tab to generate a random passphrase.

You can keep the database on a USB drive, but make sure you keep a backup in a more reliable location. You should always keep a backup of your passwords database anyway. (Or rather, your *primary* copy should be in the more reliable location.)

**Comment by ghost** [comment]
Well lol i just want to say in the paste the usb drives where a risk
Modern high quality flashdrives like lexar never have failed me till now so depending on the quality of the product i think the risk is not so great as in the old days. If you do not buy cheap crap
Unless you buy cheap stuff the risk of loosing data is much smaller now
But to be honest i never ever lost data on the flashdrives i have bought in the paste there was a trick to keep them not loosing data by refreshing the data on them. After 12 years it finally started to showed a bit degradation.

But it simply was also no longer usefull as the sizes needed became so great 
My oldest modern sticks with sizes which are usefull these days all data is still in tact after more than 8 years without having to refresh the data. I do not use them anymore as these days we are using silly big amounts sized ones 64 to 512 Gb is "normaL" now.
 
the real risk of the usb drives is that if you forget it or gets stolen is a more important factor, that why i constant pull it out even at home and put it safe as i fear i forget it in the wrong places.
Even if i go for a coffee i pull it out so that nobody can steal or can access it even though it is secured by passwords
